subsys/logging/log_core.c

```diff
@@ -533,6 +505,14 @@ void log_backend_enable(struct log_backend const *const backend,
 			void *ctx,
 			u32_t level)
 {
+	u32_t id;
+
+	id = (u32_t)((u8_t *)backend - (u8_t*)log_backend_get(0))/
+			sizeof(struct log_backend);
```

Comment:
The way C pointer arithmetic works, a simpler way would be just `backend - log_backend_get(0)`. No division needed due to pointer types. We use the same trick e.g. here:
https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/host/conn.c#L2254

Reply:
good point. fixed.